projects/components/src/info-icon/info-icon.component.ts

```diff
@@ -0,0 +1,28 @@
+import { ChangeDetectionStrategy, Component, Input, TemplateRef } from '@angular/core';
+import { IconType } from '@hypertrace/assets-library';
+import { IconSize } from '../icon/icon-size';
+
+@Component({
+  selector: 'ht-info-icon',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <ng-container *ngIf="this.info"
+      ><ht-icon icon="${IconType.Info}" [size]="this.iconSize" [htTooltip]="infoTooltipTemplate"></ht-icon>
+      <ng-template #infoTooltipTemplate>
+        <ng-container *ngTemplateOutlet="this.isInfoAString ? defaultInfoTooltipTemplate : this.info"></ng-container>
+      </ng-template>
+      <ng-template #defaultInfoTooltipTemplate> {{ this.info }}</ng-template></ng-container
+    >
+  `
+})
+export class InfoIconComponent {
+  @Input()
+  public info?: string | TemplateRef<unknown>;
```

Comment:
Similar to how we have **context** for **Tooltip**, would it make sense to have similar logic implemented?